### PR TITLE
Update installLisk.sh to not use 'latest' build

### LINKED ISSUE #145: Update installLisk.sh to handle future change to downloads.lisk.io structure
The `latest` releases are going away at some point. installLisk.sh will get the most recent version from the highest numbered version directory, by default, while also providing an option to specify the version

### PR DESCRIPTION
**Pull request by Nazgolze**
This closes #145

### CODE REVIEW

File: downloaded/installLisk.sh [hunk 1]
```diff
@@ -101,17 +101,21 @@ download_lisk() {
 		return
 	fi
 
-	LISK_VERSION=lisk-$UNAME.tar.gz
+	if [[ ! $LISK_VERSION_NUMBER ]]; then
+		LISK_VERSION_NUMBER=$(curl -s "https://downloads.lisk.io/lisk/$RELEASE/latest.txt")
```
Comment (fchavant) [inline downloaded/installLisk.sh]: Can we fail gracefuly if `latest.txt` does not exist / cannot be fetched?

File: downloaded/installLisk.sh [hunk 2]
```diff
@@ -101,17 +101,21 @@ download_lisk() {
...
+	LISK_VERSION=lisk-$LISK_VERSION_NUMBER-$UNAME.tar.gz
 
 	LISK_DIR=${LISK_VERSION%.tar.gz}
 
 	rm -f "$LISK_VERSION" "$LISK_VERSION".SHA256 &> /dev/null
 
 	echo -e "\\nDownloading current Lisk binaries: ""$LISK_VERSION"
 
-	curl --progress-bar -o "$LISK_VERSION" "https://downloads.lisk.io/lisk/$RELEASE/$LISK_VERSION"
+	curl --progress-bar -o "$LISK_VERSION" "https://downloads.lisk.io/lisk/$RELEASE/$LISK_VERSION_NUMBER/$LISK_VERSION"
```
Comment (fchavant) [inline downloaded/installLisk.sh]: Same as above: we should fail with an understandable error message if the file cannot be downloaded.

File: downloaded/installLisk.sh [hunk 3]
```diff
@@ -272,23 +276,24 @@ upgrade_lisk() {
...
+	echo "install                    -- install Lisk"
+	echo "upgrade                    -- upgrade Lisk"
+	echo " -d <DIRECTORY>            -- install location"
+	echo " -f <FILE>                 -- use a local tarball to install"
+	echo " -r <main|test|dev>        -- choose network (default: main)"
+	echo " -h                        -- rebuild instead of copying database"
+	echo " -u <URL>                  -- URL to rebuild from - Requires -h"
+	echo " -0 <yes|no>               -- force sync from 0 (default: no)"
+	echo " -c                        -- clean database after upgrade"
+	echo " -s <LISK_VERSION_NUMBER>  -- specify a version of lisk-core. ex: -s 1.0.2"
```
Comment (fchavant) [inline downloaded/installLisk.sh]: Should mention that the default is to get the latest version.

File: downloaded/installLisk.sh [hunk 4]
```diff
@@ -303,6 +320,7 @@ parse_option() {
 			u) URL="$OPTARG" ;;
 			0) SYNC="$OPTARG" ;;
 			c) CLEAN_DB="yes" ;;
+			s) LISK_VERSION_NUMBER="$OPTARG" ;;
```
Comment (fchavant) [inline downloaded/installLisk.sh]: Why `-s`? (and not, say, `-v`?)

File: downloaded/installLisk.sh [hunk 5]
```diff
@@ -101,17 +101,33 @@ download_lisk() {
...
+	LISK_VERSION=lisk-$LISK_VERSION_NUMBER-$UNAME.tar.gz
 
 	LISK_DIR=${LISK_VERSION%.tar.gz}
 
 	rm -f "$LISK_VERSION" "$LISK_VERSION".SHA256 &> /dev/null
 
 	echo -e "\\nDownloading current Lisk binaries: ""$LISK_VERSION"
 
-	curl --progress-bar -o "$LISK_VERSION" "https://downloads.lisk.io/lisk/$RELEASE/$LISK_VERSION"
+	DOWNLOAD_CHECK=$(curl -f --progress-bar -o "$LISK_VERSION" "https://downloads.lisk.io/lisk/$RELEASE/$LISK_VERSION_NUMBER/$LISK_VERSION" || echo error)
```
Comment (fchavant) [inline downloaded/installLisk.sh]: Why not check for `curl`'s return code?

File: downloaded/installLisk.sh (hunk 4 above)
Comment (Nazgolze) [reply downloaded/installLisk.sh]: I was saving `-v` for either a verbose mode or having `installLisk.sh -v` spit out its own version number, if we go down that road

File: downloaded/installLisk.sh [hunk 6]
```diff
@@ -80,6 +80,11 @@ user_prompts() {
 }
 
 download_lisk() {
+	if [ "${LISK_VERSION_NUMBER:-}" ] ; then
+		echo "Getting latest lisk version" 
+		LISK_VERSION_NUMBER=$(curl --progress-bar --fail "https://downloads.lisk.io/lisk/$RELEASE/latest.txt")
+	fi
+	LISK_VERSION=lisk-$LISK_VERSION_NUMBER-Linux-x86_64.tar.gz
```
Comment (fchavant) [inline downloaded/installLisk.sh]: `LISK_VERSION=lisk-Linux-x86_64.tar.gz` should be removed above

File: downloaded/installLisk.sh [hunk 7]
```diff
@@ -80,6 +80,11 @@ user_prompts() {
 }
 
 download_lisk() {
+	if [ "${LISK_VERSION_NUMBER:-}" ] ; then
+		echo "Getting latest lisk version" 
+		LISK_VERSION_NUMBER=$(curl --progress-bar --fail "https://downloads.lisk.io/lisk/$RELEASE/latest.txt")
```
Comment (fchavant) [inline downloaded/installLisk.sh]: Do we need `--progress-bar` for such a small file?

**Comment by Nazgolze** [comment]
sure, will squash. One sec....

**Pull request merged**